### PR TITLE
Fix crash when sawing down a barrell of a gun containing a magazine

### LINKED ISSUE #16045: Crash when trying to saw down the barrel of Saiga-12
Caught this accidentally while trying to cut some metal bars. Had Saiga-12 in the inventory.

Reproducing:
1. Have Saiga-12 with box magazine attached in the inventory
2. (a)ctivate hacksaw from inventory
3. Choose "saw down barrel"
4. Game crashes

Works as intended if magazine is not attached.

Version 4624 (0.C-12559-gdb57510)

### PR DESCRIPTION
**Pull request by codemime**
Fixes #16045

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -8493,13 +8493,13 @@ int iuse::saw_barrel( player *p, item *, bool, const tripoint& )
...
 
-    auto filter = [&]( const item& e ) {
+    auto filter = []( const item& e ) {
         if( !e.is_gun() || e.type->gun->barrel_length <= 0 ) {
             return false;
         }
         // cannot saw down barrel of gun that already has a barrel mod
-        return std::none_of( e.contents.begin(), e.contents.end(), [&]( const item& mod ) {
-            return mod.type->gunmod->location == "barrel";
+        return std::none_of( e.contents.begin(), e.contents.end(), []( const item& mod ) {
```
Comment (mugling) [inline src/iuse.cpp]: This correctly identifies the problem however storing the result of item::gunmods() and then using this for `std::none_of` is preferred to directly accessing the `contents` vector.

Comment (codemime) [inline src/iuse.cpp]: I see no advantage. It requires constructing yet another vector and doubles iteration costs, but without any sensible benefit: it doesn't reduce amount of code and actually increases it. Clarity is arguable. The only strong reason for doing so is encapsulating `contents` (or storing gunmods somewhere else in the future), but major goal of the PR was fixing the crash, not refactoring your code.

Comment (mugling) [inline src/iuse.cpp]: > doubles iteration costs

That code branch could never become a performance concern

> The only strong reason for doing so is encapsulating contents

Encapsulation is important and precisely why this crash occurred in the first place.

**Comment by codemime** [comment]
All set.

**Pull request merged**